### PR TITLE
fix: fixed case when a project with an already existing name was created

### PR DESCRIPTION
**Pull request by npetalid**

### CODE REVIEW

File: srcup/cli.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 import asyncio
 import builtins
+import sys
```
Comment (iliastsa) [inline srcup/cli.py:6]: Unused import?

File: srcup/api.py [hunk 2]
```diff
@@ -103,4 +112,4 @@ async def get_project_id(watchdog_api: str,
         if req.status == 200:
             return await req.json()
         else:
-            raise Exception("No project with name {name} exists")
\ No newline at end of file
+            return -1  # project does not exist
```
Comment (iliastsa) [inline srcup/api.py]: Returning `None` instead is a bit cleaner I think

File: srcup/api.py [hunk 3]
```diff
@@ -29,6 +29,12 @@ class Config:
         headers={"x-api-key": api_key}, json_serialize=lambda x: x.json()
     ) as session:
         print("Uploading...")
+
+        project_id = await get_project_id(watchdog_api, api_key, name)
+
+        if project_id != -1:
+            raise Exception(f"Project with name {name} already exists")
```
Comment (iliastsa) [inline srcup/api.py]: If we are to return `None` (see comment below) this will need to be an `is None` check.

File: srcup/cli.py (hunk 1 above)
Comment (npetalid) [reply srcup/cli.py:6]: isn't it used in sys.exit()?

File: srcup/api.py (hunk 2 above)
Comment (npetalid) [reply srcup/api.py]: Done

File: srcup/api.py (hunk 3 above)
Comment (npetalid) [reply srcup/api.py]: Done

File: srcup/cli.py (hunk 1 above)
Comment (iliastsa) [reply srcup/cli.py:6]: You are right, missed that!

File: srcup/api.py (hunk 3 above)
Comment (iliastsa) [reply srcup/api.py]: I don't t see the `is not None` check

File: srcup/api.py [hunk 4]
```diff
@@ -80,14 +89,14 @@ class Config:
...
+            error = await req.text()
+            raise Exception(error)
 
 
 async def get_project_id(watchdog_api: str,
     api_key: str,
-    owner_username: str,
     name: str,
+    owner_username: str = '',
 ) -> int:
```
Comment (iliastsa) [inline srcup/api.py]: `int | None`

**Review by iliastsa — APPROVED** [review]

**Pull request merged**